library/x509_create.c

```diff
@@ -382,6 +382,10 @@ int mbedtls_x509_set_extension(mbedtls_asn1_named_data **head, const char *oid,
 {
     mbedtls_asn1_named_data *cur;
 
+    if (val_len > (SIZE_MAX  - 1)) {
```

Comment:
(1) Can we have a comment to explain why this value of `val_len` is unacceptable
(2) Since this only value satisfying this inequality is `SIZE_MAX` itself, why not use `== SIZE_MAX`?